src/coreclr/ilasm/sha256.h

```diff
@@ -0,0 +1,145 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//*****************************************************************************
+// sha256.h
+//
+
+//
+// contains implementation of sha256 hash algorithm
+//
+//*****************************************************************************
+#ifndef __sha256__h__
+#define __sha256__h__
+
+#ifdef _WIN32
+inline HRESULT Sha256Hash(BYTE* pSrc, DWORD srcSize, BYTE* pDst, DWORD dstSize)
+{
+    NTSTATUS status;
+
+    BCRYPT_ALG_HANDLE   algHandle = NULL;
+    BCRYPT_HASH_HANDLE  hashHandle = NULL;
+
+    BYTE    hash[32]; // 256 bits
+    DWORD   hashLength = 0;
+    DWORD   resultLength = 0;
+    status = BCryptOpenAlgorithmProvider(&algHandle, BCRYPT_SHA256_ALGORITHM, NULL, BCRYPT_HASH_REUSABLE_FLAG);
+    if(!NT_SUCCESS(status))
+    {
+        goto cleanup;
+    }
+    status = BCryptGetProperty(algHandle, BCRYPT_HASH_LENGTH, (PBYTE)&hashLength, sizeof(hashLength), &resultLength, 0);
+    if(!NT_SUCCESS(status))
+    {
+        goto cleanup;
+    }
+    if (hashLength != 32)
+    {
+        status = STATUS_NO_MEMORY;
+        goto cleanup;
+    }
+    status = BCryptCreateHash(algHandle, &hashHandle, NULL, 0, NULL, 0, 0);
+    if(!NT_SUCCESS(status))
+    {
+        goto cleanup;
+    }
+
+    status = BCryptHashData(hashHandle, pSrc, srcSize, 0);
+    if(!NT_SUCCESS(status))
+    {
+        goto cleanup;
+    }
+
+    status = BCryptFinishHash(hashHandle, hash, hashLength, 0);
+    if(!NT_SUCCESS(status))
+    {
+        goto cleanup;
+    }
+
+    if (dstSize < hashLength)
+    {
+    	memcpy(pDst, hash, dstSize);
+    }
+    else
+    {
+    	memcpy(pDst, hash, hashLength);
+    }
+
+    status = S_OK;
+
+cleanup:
+    if (NULL != hashHandle)
+    {
+         BCryptDestroyHash(hashHandle);
+    }
+    if(NULL != algHandle)
+    {
+        BCryptCloseAlgorithmProvider(algHandle, 0);
+    }
+    return status;
+}
+#elif defined(__APPLE__)
+#include <CommonCrypto/CommonCrypto.h>
+#include <CommonCrypto/CommonDigest.h>
+
+inline HRESULT Sha256Hash(BYTE* pSrc, DWORD srcSize, BYTE* pDst, DWORD dstSize)
+{
+    BYTE hash[32];
+    CC_SHA256(pSrc, (CC_LONG)srcSize, hash);
+
+    if (dstSize < CC_SHA256_DIGEST_LENGTH)
+    {
+        memcpy(pDst, hash, dstSize);
+    }
+    else
+    {
+        memcpy(pDst, hash, CC_SHA256_DIGEST_LENGTH);
+    }
+
+    return S_OK;
+}
+#elif defined(__linux__)
```

Comment:
This can just `#else` as `CryptoNative_EvpDigestOneShot` is available on any Unix.